src/actions/genSale/purchaseDomains.ts

```diff
@@ -0,0 +1,88 @@
+import * as ethers from "ethers";
+import { getSaleStatus } from ".";
+import { GenSale } from "../../contracts/types";
+import { Claim, Mintlist, Maybe, GenSaleStatus } from "../../types";
+
+const abi = ["function masterCopy() external view returns (address)"];
+
+const errorCheck = (condition: boolean, errorMessage: string) => {
```

Comment:
let's make a habit of writing general helpers in their own respective locations and reuse the same ones everywhere.
I don't like the current process of copying the same code for every single operation. the current function is present 3 times already in this repo.
let's approach our code in a more abstract way. at least starting with little things like this one.

Reply:
I moved this errorCheck function to the helpers folder